network/tcp.go

```diff
@@ -284,7 +284,15 @@ func handleError(err error) error {
 		return ErrTimeout
 	}
 
-	log.Errorf("Unknown error caught: %s", err.Error())
+	if strings.Contains(err.Error(), "bad certificate") {
+		// This is a weak indication of the accidental
+		// onet 3.2.4/3.2.5 backwards incompatiblity, so explain
+		// the problem to help the user.
+		log.Errorf("Unknown error caught: %s %v", err.Error(),
+			"(This may be caused by Onet > v3.2.4 talking to Onet <= v3.2.4. See onet #657 for more info.)")
```

Comment:
```suggestion
			"(This may be caused by Onet > v3.2.4 talking to Onet <= v3.2.4. See https://github.com/dedis/onet/issues/657 for more info.)")
```